packages/lib/src/vis/heatmap/HeatmapMesh.tsx

```diff
@@ -174,21 +110,25 @@ function HeatmapMesh(props: Props) {
     return new DataTexture(colors, CMAP_SIZE, 1, RGBFormat, UnsignedByteType);
   }, [colorMap, invertColorMap]);
 
-  const scaleUniforms = isScaleType(scaleType)
-    ? SCALE_SHADER[scaleType].uniforms(domain)
-    : SCALE_SHADER[ScaleType.Gamma].uniforms(domain, scaleType[1] as number);
+  const [scaleType, gammaExponent] = isScaleType(visScaleType)
+    ? [visScaleType]
+    : visScaleType;
+
+  const scaledDomain = scaleDomain(domain, scaleType as ScaleType);
 
   const shader = {
     uniforms: {
       data: { value: dataTexture },
-      withAlpha: { value: alphaValues ? 1 : 0 },
-      alpha: { value: alphaTexture },
       colorMap: { value: colorMapTexture },
-      alphaMin: { value: alphaDomain?.[0] },
+      min: { value: scaledDomain[0] },
+      oneOverRange: { value: 1 / (scaledDomain[1] - scaledDomain[0]) },
+      gammaExponent: { value: gammaExponent },
+      alpha: { value: alphaTexture },
+      withAlpha: { value: alphaValues ? 1 : 0 },
+      alphaMin: { value: alphaDomain[0] },
       oneOverAlphaRange: {
-        value: alphaDomain && 1 / (alphaDomain[1] - alphaDomain[0]),
+        value: 1 / (alphaDomain[1] - alphaDomain[0]),
       },
```

Comment:
The alpha domain is now always passed. It's not used in the shader when `withAlpha` is `0`, so it doesn't change anything. It just removes a couple of ternary/optional chaining. It was simplifying a lot more things when I had to normalize the domains when dealing with `uint8`, but I still think it's worth it. I'm hoping that the performance impact of this is negligible, but please correct me if I'm wrong and I'll revert.

Reply:
I don't see how this will impact performance: the `uniform` was always there even if it could be `undefined` before.